From: Michal Schmidt <mschmidt@redhat.com>
Date: Fri, 3 Sep 2010 16:43:38 -0400
Subject: [net] bnx2x: return -EINVAL for unsupported flags
Message-id: <20100903164338.25415.30887.stgit@brian.englab.brq.redhat.com>
Patchwork-id: 28105
O-Subject: [RHEL5.6 BZ572012 PATCH 30/46] bnx2x: Make ethtool_ops::set_flags()
	return -EINVAL for unsupported flags
Bugzilla: 572012
RH-Acked-by: David S. Miller <davem@redhat.com>
RH-Acked-by: Stanislaw Gruszka <sgruszka@redhat.com>

from upstream 97d1935a61b7fe7a65f98f154c7f3301cfe746f3

RHEL: in #if 0, no effect

diff --git a/drivers/net/bnx2x_main.c b/drivers/net/bnx2x_main.c
index 86a71c6..9890c27 100644
--- a/drivers/net/bnx2x_main.c
+++ b/drivers/net/bnx2x_main.c
@@ -11013,7 +11013,7 @@ static int bnx2x_set_flags(struct net_device *dev, u32 data)
 	int rc = 0;
 
 	if (data & ~(ETH_FLAG_LRO | ETH_FLAG_RXHASH))
-		return -EOPNOTSUPP;
+		return -EINVAL;
 
 	if (bp->recovery_state != BNX2X_RECOVERY_DONE) {
 		printk(KERN_ERR "Handling parity error recovery. Try again later\n");
